Improving loading states of country stat pills

jdan opened the pull request:
(My first time using Svelte!) This changeset throws all json data loading for country pills behind a Promise.all for parallelization. It then introduces a new `jsonDataLoading` state so not even the submit button appears while loading. This should reduce some flickering as the data loads in. There's likely room for reducing more layout jank here, but this is just a start.

Before

https://github.com/user-attachments/assets/8ec6ec7b-9800-420b-ac23-6bd02c5df748

After

https://github.com/user-attachments/assets/967855d0-2316-4d87-85f1-1d59d392ee64

therealPaulPlay commented:
Hey, thanks for your contribution! I like the idea, but I'm not sure if this makes sense this way... In the after example, there is more layout shift than in the before example.

jdan commented:
Yeah I think 0d95fc7 + 9756778 hide that entire bar but in reality should keep it but with a gray bg

jdan commented:
@therealPaulPlay I've changed it so that the bar itself maintains height before and after the json loads. I've updated the "after" video in the description. Lemme know what you think!

therealPaulPlay commented:
Thank you, looks less jittery now. I feel like this isn't formatted correctly (diff log shows a lot of formatting changes), did you run prettier?

jdan commented:
@therealPaulPlay the formatting is the same as the surrounding code. there are changed lines because I wrapped some markup in an if statement, but the indentation is the same

![image](https://github.com/user-attachments/assets/47824799-217a-40cf-ae73-70b7c4c8c22a)

therealPaulPlay commented:
Right, I see, thanks a ton!!

therealPaulPlay commented:
By the way, this broke the video loading:

```javascript
			videoSources = await fetch("/json/country-data/country-videos.json").then((response) =>
				response.json().then((response) => response[countryName]),
			);
```

used to be a different assignment than the rest. Will fix that.

jdan commented:
😭 sorry about that

On Thu, Apr 3, 2025 at 4:32 PM PaulPlay ***@***.***> wrote:

> By the way, this broke the video loading:
>
> 			videoSources = await fetch("/json/country-data/country-videos.json").then((response) =>
> 				response.json().then((response) => response[countryName]),
> 			);
>
> used to be a different assignment than the rest. Will fix that.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/therealPaulPlay/OpenGuessrEducation/pull/37#issuecomment-2776855570>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACGEJDAUR226SN2A7DRAAL2XWLG7AVCNFSM6AAAAABYQDM7ZSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZWHA2TKNJXGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: therealPaulPlay]*therealPaulPlay* left a comment
> (therealPaulPlay/OpenGuessrEducation#37)
> <https://github.com/therealPaulPlay/OpenGuessrEducation/pull/37#issuecomment-2776855570>
>
> By the way, this broke the video loading:
>
> 			videoSources = await fetch("/json/country-data/country-videos.json").then((response) =>
> 				response.json().then((response) => response[countryName]),
> 			);
>
> used to be a different assignment than the rest. Will fix that.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/therealPaulPlay/OpenGuessrEducation/pull/37#issuecomment-2776855570>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACGEJDAUR226SN2A7DRAAL2XWLG7AVCNFSM6AAAAABYQDM7ZSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZWHA2TKNJXGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
